plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/ProjectStatePersister.java

```diff
@@ -485,10 +497,32 @@ private File getDataFile(IProjectConfig project) {
 		return file;
 	}
 
+	private String makeRelative(URI baseURI, URI absoluteURI) {
+		if (absoluteURI.isFile()) {
+			URI relativeURI = absoluteURI.deresolve(baseURI);
+			return relativeURI.toString();
+		}
+		return absoluteURI.toString();
+	}
+
+	private URI makeAbsolute(URI baseURI, String relativeString) {
+		if (relativeString.contains(":/")) {
+			return URI.createURI(relativeString);
+		}
+		URI relativeURI = URI.createFileURI(relativeString);
+		URI absoluteURI = relativeURI.resolve(baseURI);
+		return new FileURI(absoluteURI).toURI();
```

Comment:
Why converting it to a FileURI and back to an URI?

Reply:
FileURI performs several assertions that I wanted to reuse here.